var/spack/repos/builtin/packages/augustus/package.py

```diff
@@ -0,0 +1,74 @@
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+import shutil
+
+
+class Augustus(MakefilePackage):
+    """AUGUSTUS is a program that predicts genes in eukaryotic
+       genomic sequences"""
+
+    homepage = "http://bioinf.uni-greifswald.de/augustus/"
+    url      = "http://bioinf.uni-greifswald.de/augustus/binaries/augustus-3.2.3.tar.gz"
+
+    version('3.2.3', 'b8c47ea8d0c45aa7bb9a82626c8ff830')
+
+    depends_on('bamtools')
+    depends_on('gsl')
+    depends_on('boost')
+    depends_on('zlib')
+
+    def edit(self, spec, prefix):
+        with working_dir(os.path.join(self.build_directory,
+                         'auxprogs', 'filterBam', 'src')):
+            makefile = FileFilter('Makefile')
+            makefile.filter('BAMTOOLS = .*', 'BAMTOOLS = %s' % self.spec[
+                            'bamtools'].prefix)
+            makefile.filter('INCLUDES = *',
+                            'INCLUDES = -I$(BAMTOOLS)/include/bamtools ')
+            makefile.filter('LIBS = -lbamtools -lz',
+                            'LIBS = $(BAMTOOLS)/lib/bamtools'
+                            '/libbamtools.a -lz')
+        with working_dir(os.path.join(self.build_directory,
+                         'auxprogs', 'bam2hints')):
+            makefile = FileFilter('Makefile')
+            makefile.filter('# Variable definition',
+                            'BAMTOOLS = %s' % self.spec['bamtools'].prefix)
+            makefile.filter('INCLUDES = /usr/include/bamtools',
+                            'INCLUDES = $(BAMTOOLS)/include/bamtools')
+            makefile.filter('LIBS = -lbamtools -lz',
+                            'LIBS = $(BAMTOOLS)/lib/bamtools'
+                            '/libbamtools.a -lz')
+
+    def install(self, spec, prefix):
+        shutil.copytree('bin', os.path.join(self.spec.prefix, 'bin'))
+        shutil.copytree('config', os.path.join(self.spec.prefix, 'config'))
+        shutil.copytree('scripts', os.path.join(self.spec.prefix, 'scripts'))
```

Comment:
Spack also has a builtin function called `install_tree` if you don't want to import shutil. It works the same way.